src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,379 @@
+                        }
+                        disableDown={
+                            currentIsTime &&
+                            !isValidValue(currentSegmentValue - 1, currentSegment as any, ampm)
+                        }
+                        onUp={this.createStepperHandler(+1)}
+                        onDown={this.createStepperHandler(-1)}
+                    />
+                }
+                {isTouch &&
+                    <label className="label">
+                        <input
+                            className="native-input"
+                            type="time"
+                            name={name}
+                            aria-label={label}
+                            value={this.getValue()}
+                            disabled={disabled}
+                            onFocus={this.onNantiveInputFocus}
+                            onBlur={this.onInputBlur}
+                            onChange={this.onNativeInputChange}
+                        />
+                    </label>
+                }
+            </div>
+        );
+    }
+
+    private getInitialSegments(props: Props): Pick<State, 'hh' | 'mm' | 'ampm' | 'inputValue'> {
+        const segments = propsValueToSegments(props.value, props.format);
+        return {
+            ...segments,
+            inputValue: segmentsToInputValue(segments)
+        };
+    }
+
+    private getValue(): string {
+        const {hh, mm, ampm} = this.state;
+        return [
+            to24(hh || 0, ampm),
+            mm || 0
+        ].map(String).map(pad2).join(':');
+    }
+
+    private select(segment: Segment): boolean {
+        const indexes = selectionIndexes[segment];
+        const input = this.input!;
+        const cursorPosition = input.selectionStart;
+        input.selectionStart = indexes[0];
+        input.selectionEnd = indexes[1];
+        this.setState({currentSegment: segment});
+        return input.selectionStart !== input.selectionEnd;
+    }
+    private moveSelection(step: number): boolean {
+        const segments = Object.keys(selectionIndexes) as Segment[];
+        const index = segments.indexOf(this.state.currentSegment!);
+        const nextSegment = segments[index + step];
+        if (nextSegment) {
+            return this.select(nextSegment);
+        }
+        return false;
+    }
+
+    private updateSegmentValue(name: Segment, value: number | Ampm): void {
+        const {hh, mm, ampm} = this.state;
+        this.setState({
+            [name as any]: value,
+            inputValue: segmentsToInputValue({
+                hh, mm, ampm,
+                [name]: value
+            })
+        }, () => {
+            this.input!.focus();
+            this.select(name);
+            this.commit();
+        });
+    }
+
+    private toggleAmpm(): void {
+        this.updateSegmentValue('ampm', 1 - this.state.ampm);
+    }
+
+    private changeValue(step: number): void {
```

Comment:
only here I've got and idea why `1 - ampm`, can you somehow run different handlers for changing ampm and hh/mm, because it is weird how it is now.